bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -262,6 +258,32 @@ public void partClosed(IWorkbenchPart part) {
 		public void partOpened(IWorkbenchPart part) {
 			// Do nothing
 		}
+
+		private void adaptToPartActivationChange() {
+			if (getShell() == null || targetPart.getSite().getShell() == null) {
```

Comment:
`targetPart.getSite().getShell() == null` that is same as `getShell()`. Instead, check for the `targetPart.getSite().getPart() == null`

Reply:
*No, that is not the same as `getShell()`* as this code is ran from inside of a Dialog. `getShell()` returns the shell of the Dialog, and `targetPart.getSite().getShell()` returns the the shell which contains the part we are targetting.

Sure, I can change the check for whether the part is null. Please refer to my comment down-below.

Reply:
Pleas ehcekc whether my change in https://github.com/eclipse-platform/eclipse.platform.ui/commit/ad63e43557ee5c01ce8857d21d4dc5436905f9bb is addressing your concerns